src/main/webapp/app/tasks/task-list/task-list.component.ts

```diff
@@ -20,9 +23,38 @@ export class TaskListComponent {
 
   constructor(@Inject('TaskService') private taskService: TaskService) { }
 
+  update(task: Task): void {
+    this.taskService.update(task.id, task)
+      // This is a great method to unsubscribe from observable when it destoyed
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.sortPendingTasks(this.tasks);
+    });
+  }
+
   delete(task: Task): void {
-    this.taskService.delete(task.id).subscribe(() => {
-      this.deleted.emit(task);
+    this.taskService.delete(task.id)
+      // This is a great method to unsubscribe from observable when it destoyed
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.deleted.emit(task);
     });
   }
+
+  clearCompletedTasks(tasks: Task[]): void {
+    // Optimistic approach
+    const currentTasks = tasks.filter( task => task.isCompleted === false );
```

Comment:
As said if i name the service method "clearCompletedTasks" i would assume that the service is taking care of filtering out all completed tasks. Another solution would be to filter for completed tasks and add a service method "deleteAll" or something. I would prefer the first solution.

Reply:
great input 👍 , Thanks